components/browser/engine-gecko-nightly/src/test/java/mozilla/components/browser/engine/gecko/GeckoEngineSessionTest.kt

```diff
@@ -1902,15 +1902,105 @@ class GeckoEngineSessionTest {
     }
 
     @Test
-    fun `Handle new window load requests`() {
-        GeckoEngineSession(mock(), geckoSessionProvider = geckoSessionProvider)
+    fun `onLoadRequest will intercept new window load requests`() {
+        val engineSession = GeckoEngineSession(mock(),
+            geckoSessionProvider = geckoSessionProvider)
+
         captureDelegates()
 
-        val result = navigationDelegate.value.onLoadRequest(geckoSession,
-                mockLoadRequest("sample:about", null, GeckoSession.NavigationDelegate.TARGET_WINDOW_NEW))
+        var observedUrl: String? = null
+        var observedIntent: Intent? = null
 
-        assertNotNull(result)
-        assertEquals(result!!.poll(0), AllowOrDeny.ALLOW)
+        engineSession.settings.requestInterceptor = object : RequestInterceptor {
+            override fun interceptsAppInitiatedRequests() = true
+
+            override fun onLoadRequest(
+                engineSession: EngineSession,
+                uri: String,
+                hasUserGesture: Boolean,
+                isSameDomain: Boolean,
+                isRedirect: Boolean,
+                isDirectNavigation: Boolean
+            ): RequestInterceptor.InterceptionResponse? {
+                return when (uri) {
+                    "sample:about" -> RequestInterceptor.InterceptionResponse.AppIntent(mock(), "result")
+                    else -> null
+                }
+            }
+        }
+
+        engineSession.register(object : EngineSession.Observer {
+            override fun onLaunchIntentRequest(
+                url: String,
+                appIntent: Intent?
+            ) {
+                observedUrl = url
+                observedIntent = appIntent
+            }
+        })
+
+        navigationDelegate.value.onLoadRequest(
+            mock(), mockLoadRequest("sample:about", null,
+            GeckoSession.NavigationDelegate.TARGET_WINDOW_NEW, triggeredByRedirect = true)
+        )
+
+        assertNotNull(observedIntent)
+        assertEquals("result", observedUrl)
+
+        navigationDelegate.value.onLoadRequest(
+            mock(), mockLoadRequest("sample:about", null,
+            GeckoSession.NavigationDelegate.TARGET_WINDOW_NEW, triggeredByRedirect = false)
+        )
+
+        assertNotNull(observedIntent)
+        assertEquals("result", observedUrl)
+    }
+
+    @Test
+    fun `onLoadRequest will not intercept new window blank page load requests`() {
```

Comment:
Hm this isn't quite right. What you're testing here is that we can open new windows e.g. (target="_blank" pages). These aren't blank pages, just pages we want to open in a new tab.

The test is otherwise right. I would use a different example than `about:blank` and a different name e.g.

```onLoadRequest allows new window requests if not intercepted``` (or similar) :)

Reply:
Sure, I'll add another URL in the test as part of the update as well.  Thanks,